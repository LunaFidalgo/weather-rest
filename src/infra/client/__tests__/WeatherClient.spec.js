const WeatherClient = require('../WeatherClient')
const nock = require('nock')
const mock = require('./__mocks__/mock_data')

describe('Api weather client', () => {
  const logger = {
    info: jest.fn()
  }
  const config = {
    weather: {
      apiEndpoint: 'api.openweathermap.org/data/2.5/weather?q=Madrid&appid=',
      apiKey: '12345'
    }
  }

  test('client instance', () => {
    const weatherClient = new WeatherClient({ logger, config })

    expect(weatherClient).toBeDefined()
  })

  test('Construct url', () => {
    const weatherClient = new WeatherClient({ logger, config })

    expect(weatherClient._url()).toMatchSnapshot()
  })

  test('Get weather', async () => {
    const weatherClient = new WeatherClient({ logger, config })

    nock(`http://api.openweathermap.org`)
      .get('/data/2.5/weather')
      .query({
        q: 'Madrid',
        appid: '12345'
      })
      .reply(200, mock.mockResponseMadrid())
    nock.disableNetConnect()

    const resp = await weatherClient.getWeather()
    nock.enableNetConnect()

    expect(resp).toMatchSnapshot()
  })
})
