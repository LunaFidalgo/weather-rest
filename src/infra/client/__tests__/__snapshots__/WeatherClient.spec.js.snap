// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Api weather client Construct url 1`] = `"http://api.openweathermap.org/data/2.5/weather?q=Madrid&appid=12345"`;

exports[`Api weather client Get weather 1`] = `
Object {
  "base": "stations",
  "clouds": Object {
    "all": 20,
  },
  "cod": 200,
  "coord": Object {
    "lat": 40.42,
    "lon": -3.7,
  },
  "dt": 1528660800,
  "id": 3117735,
  "main": Object {
    "humidity": 72,
    "pressure": 1013,
    "temp": 290.43,
    "temp_max": 291.15,
    "temp_min": 289.15,
  },
  "name": "Madrid",
  "sys": Object {
    "country": "ES",
    "id": 5505,
    "message": 0.0022,
    "sunrise": 1528605850,
    "sunset": 1528659891,
    "type": 1,
  },
  "visibility": 10000,
  "weather": Array [
    Object {
      "description": "few clouds",
      "icon": "02n",
      "id": 801,
      "main": "Clouds",
    },
  ],
  "wind": Object {
    "deg": 240,
    "speed": 4.1,
  },
}
`;
